src/Trip/index.ts

```diff
@@ -0,0 +1,68 @@
+import { stringify } from 'query-string';
+
+import { getPlaceDetailedBatch } from '../Places/index';
+import { Place } from '../Places/Place';
+import { get } from '../Xhr';
+import {
+	mapTripDetailedApiResponseToTrip,
+	mapTripListApiResponseToTripsList
+} from './Mapper';
+import { Day, ItineraryItem, Trip } from './Trip';
+
+export {
+	Trip
+};
+
+export async function getTrips(dateFrom: string, dateTo: string): Promise<Trip[]> {
+	const apiResponse = await get('trips/list?' + stringify({
+		date_from: dateFrom,
+		date_to: dateTo
+	}));
+	if (!apiResponse.data.hasOwnProperty('trips')) {
+		throw new Error('Wrong API response');
+	}
+	return mapTripListApiResponseToTripsList(apiResponse);
+}
+
+export async function getTripDetailed(id: string): Promise<Trip> {
+	const apiResponse = await get('trips/' + id);
+	if (!apiResponse.data.hasOwnProperty('trip')) {
+		throw new Error('Wrong API response');
+	}
+	const tripWithoutPlaces: Trip = mapTripDetailedApiResponseToTrip(apiResponse);
+
+	if (tripWithoutPlaces.days) {
+		return await addPlacesToTrip(tripWithoutPlaces);
+	}
+
+	return tripWithoutPlaces;
+}
+
+async function addPlacesToTrip(tripWithoutPlaces: Trip): Promise<Trip> {
+	const placesGuids: string[] = getPlacesGuidsFromTrip(tripWithoutPlaces);
+	return putPlacesToTrip(tripWithoutPlaces, await getPlaceDetailedBatch(placesGuids, '300x300'));
+}
+
+export function getPlacesGuidsFromTrip(trip: Trip): string[] {
+	if (!trip.days) {
+		return [];
+	}
+
+	const initAcc: string[] = [];
+	return trip.days.reduce((acc, day: Day): string[] => ([
+		...acc,
+		...day.itinerary.map((itineraryItem: ItineraryItem): string => (itineraryItem.placeId))
+	]), initAcc);
+}
+
+function putPlacesToTrip(trip: Trip, places: Place[]): Trip {
```

Comment:
I would move this function into mappers file. I think it is similar to mapTripwhere we pass trip and days. Also it can be better for readability (not chaining private methods).